ext/cuda/remapping_distributed.jl

```diff
@@ -67,32 +71,26 @@ function set_interpolated_values_kernel!(
     num_fields = length(field_values)
 
     hindex = (blockIdx().x - Int32(1)) * blockDim().x + threadIdx().x
-    vindex = (blockIdx().y - Int32(1)) * blockDim().y + threadIdx().y
-    findex = (blockIdx().z - Int32(1)) * blockDim().z + threadIdx().z
 
     totalThreadsX = gridDim().x * blockDim().x
-    totalThreadsY = gridDim().y * blockDim().y
-    totalThreadsZ = gridDim().z * blockDim().z
 
     _, Nq = size(I1)
     CI = CartesianIndex
-    for i in hindex:totalThreadsX:num_horiz
-        h = local_horiz_indices[i]
-        for j in vindex:totalThreadsY:num_vert
-            v_lo, v_hi = vert_bounding_indices[j]
-            A, B = vert_interpolation_weights[j]
-            for k in findex:totalThreadsZ:num_fields
-                if i ≤ num_horiz && j ≤ num_vert && k ≤ num_fields
-                    out[i, j, k] = 0
-                    for t in 1:Nq, s in 1:Nq
-                        out[i, j, k] +=
-                            I1[i, t] *
-                            I2[i, s] *
-                            (
-                                A * field_values[k][CI(t, s, 1, v_lo, h)] +
-                                B * field_values[k][CI(t, s, 1, v_hi, h)]
-                            )
-                    end
+    for k in 1:num_fields
+        for i in hindex:totalThreadsX:num_horiz
+            h = local_horiz_indices[i]
+            for j in 1:num_vert
+                v_lo, v_hi = vert_bounding_indices[j]
+                A, B = vert_interpolation_weights[j]
+                out[i, j, k] = 0
+                for t in 1:Nq, s in 1:Nq
```

Comment:
Changing the loop order, here, could have a notable impact on performance. I'm fine with first simplifying the ranges, and then maybe just update the threading pattern / parallelism.

Reply:
I changed the order precisely because I expect this to be faster because retrieving values from different fields is not the most efficient memory access pattern, so I moved the field index to the outermost loop.

In my tests, this does not degrade performance (it improves them very slightly).

But I can remove this change from this PR if you prefer

Reply:
Ok, that makes sense. I still think that the best path forward is to improve the parallelism.

Reply:
I opened #2159, to give an example of how I think this should be refactored, and how we should improve the parallelism.

Reply:
I didn't unify the indices because doing what I did in this PR was an easy fix, whereas thinking about how to do indices properly would have required more work.

Also, it's not immediately clear to me that #2159 will be better because every thread has to read the various arrays that are passed in, whereas this PR reduces some of the reads because some values are reused (e.g., `local_horiz_indices[i]`). The parallelization is over horizontal points, and for a typical output, this is order of 10^4-10^5 points, which should be plenty of threads to expose enough work to the GPU.

But if you want to go ahead and finish #2159, I'd be happy with it.